packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+import VaInputWrapper from '../va-input/components/VaInputWrapper.vue'
+import VaIcon from '../va-icon/VaIcon.vue'
+import VaButton from '../va-button/VaButton.vue'
+import { omit, pick } from 'lodash-es'
+
+const { createEmits: createInputEmits, createListeners: createInputListeners } = useEmitProxy(
+  ['change'],
+)
+
+const { createEmits: createFieldEmits, createListeners: createFieldListeners } = useEmitProxy([
+  { listen: 'click-prepend', emit: 'click:decrease-button' },
+  { listen: 'click-append', emit: 'click:increase-button' },
+  { listen: 'click-prepend-inner', emit: 'click:decrease-icon' },
+  { listen: 'click-append-inner', emit: 'click:increase-icon' },
+])
+
+export default defineComponent({
+  name: 'VaCounter',
+
+  components: { VaInputWrapper, VaIcon, VaButton },
+
+  props: {
+    ...useFormProps,
+    ...useStatefulProps,
+    // input
+    modelValue: { type: [String, Number] as PropType<string | number>, default: 0 },
+    manualInput: { type: Boolean as PropType<boolean>, default: false },
+    stateful: { type: Boolean as PropType<boolean>, default: false },
+    min: { type: Number as PropType<number>, default: undefined },
+    max: { type: Number as PropType<number>, default: undefined },
+    step: { type: Number as PropType<number>, default: 1 },
+    label: { type: String as PropType<string>, default: '' },
+    // hint
+    messages: { type: [Array, String] as PropType<string[] | string>, default: () => [] },
+    // style
+    width: { type: [String, Number] as PropType<string | number>, default: '160px' },
+    color: { type: String as PropType<string>, default: 'primary' },
+    outline: { type: Boolean as PropType<boolean> },
+    bordered: { type: Boolean as PropType<boolean> },
+    // icons & buttons
+    increaseIcon: { type: String as PropType<string>, default: 'add' },
+    decreaseIcon: { type: String as PropType<string>, default: 'remove' },
+    buttons: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: true },
+    rounded: { type: Boolean as PropType<boolean>, default: false },
+    margins: { type: [String, Number] as PropType<string | number>, default: '4px' },
+    textColor: { type: String as PropType<string | undefined>, default: undefined },
+  },
+
+  emits: [
+    'update:modelValue',
+    ...createInputEmits(),
+    ...createFieldEmits(),
+    ...useFocusEmits,
+  ],
+
+  inheritAttrs: false,
+
+  setup (props, { emit, attrs }) {
+    const input = ref<HTMLInputElement | HTMLDivElement | undefined>()
+
+    const {
+      isFocused,
+      // will be useful when we resolve problem with 'withConfigTransport'
+      focus,
+      blur,
+    } = useFocus(input, emit)
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const setCountInput = (event: Event) => {
+      valueComputed.value = +(event.target as HTMLInputElement)?.value
+    }
+
+    const setCountChange = (event: Event) => {
```

Comment:
Two handlers for same user-event? How about to have debounced `calculateCounterValue`? Or, we can just remove input handler.

Reply:
The `setCountInput` handler is required for manually entering values (props `manual-input`)